### ISSUE TITLE
Dynasty Power Rankings

### ISSUE THREAD

**Issue by lilkeeps11** [nmelhado/league-page#70]
I just copied and pasted my same post from reddit....

I've been maintaining records for my league for going on year 13 now and I have it all maintained on a google sheet (here is an old mock up I've shared) that obviously doesn't look anywhere near as nice as this (i'm not a coder by any means). But maybe there's some ideas that you can take from it and use on the site. Some other records we keep track of that I didn't see for you are Closest Matchups, Biggest blowouts, Highest/Lowest Scoring Matching ups, and Longest Winning/Losing Streaks. I saw Dynasty Power Rankings on your road map, I have a tab in the sheet that does just that with a formula I put together. It normalizes the following variables and then weights them with the following:

Championships Won - 25%

Lifetime Win% - 20%

Championship Appearances - 15%

Average Points per Week - 15%

Playoff Appearances - 10%

Season Volatility (number that compares a teams best and worst seasons) - 10%

Ave likelihood to win - 5%

But I love the Lineup IQ and would love to factor that in too (maybe a 5 or 10% weight). Anyways, feel free to DM me if you have any questions about any of this. Would love to help out with the project in any way, but it'd probably be in an idea stand point than the actual coding since what you've done is already way beyond me.

**Comment by lilkeeps11** [comment]
for your idea about factoring dynasty adp.....are you talking about accounting for, for example, all the players on my team and their ADPs, compared to all my leaguemates' teams and their players ADPs, to essentially account for current team quality?

**Comment by lilkeeps11** [comment]
I do like that idea of somehow looking at the current players, my original formula didn't include that because that league is just a 2 player keeper league not a dynasty. But thinking about how to bring that in, I have 2 questions that come to mind.
1) are we pulling dynasty ADP from sleeper? or where are we finding that
2) are we sure that we'd want ADP to be the measure? ADP essentially is the average of crowd sourcing. I wonder if we could leverage, for example, FantasyPros Dynasty Expert Consensus Rankings (ECR). I tend to like ECR over ADP to evaluate players

**Comment by nmelhado** [comment]
The idea would be to use sleeper adp. I know that's not perfect, but I'm trying to avoid any paid APIs to make it both simple and keep everything free. Ecr would be better, but I don't know of any free resources

**Comment by lilkeeps11** [comment]
yeah good call. I didn't think about the free-ness aspect. Well as far as using the ADP, I might have something of use. Granted its specific to one of my leagues (10 team, single QB, half PPR), but maybe could be extrapolated out to other leagues?

Since 2007 I've tracked a each year ADPs (from https://fantasyfootballcalculator.com/adp/standard/10-team/all) against end of season fantasy points (from https://www.pro-football-reference.com/years/2020/fantasy.htm but recalculating the stats to my league scoring). Technically I used a calculated VBD value instead of actual fantasy points to kinda normalize it across positions. But by averaging it over the past 13 years, I've essentially developed a formula to approximate the assumed VBD value of each draft pick (based on the average performance of that past pick number). It's VERY rough but at least it's something and maybe this could be used to then give "value" to each ADP position from the Sleeper ADP?

**Comment by nmelhado** [comment]
This is fantastic! This will definitely help me build dynasty power rankings. Thank you for all of this. I'm going on paternity leave soon, so major developments will be on hold. But when I start working on the website again (November-ish), this will be on the top of the list!

**Comment by lilkeeps11** [comment]
Oh awesome. Congrats man! Well, if it helps at all here's the data I've compiled. 

[its on this Google sheet](https://docs.google.com/spreadsheets/d/10FGWajqfSkcGf5WsqgR7bACuF8EIItlTV50_y9ZDeQ0/edit)

**Comment by nmelhado** [comment]
Partially filled in PR https://github.com/nmelhado/league-page/pull/75/files

Features filled:

- [x] Closest Matchups
- [x] Biggest blowouts
- [ ] Highest/Lowest Scoring Matching ups
- [ ] Longest Winning/Losing Streaks
- [ ] Dynasty rankings
